cmd/kubectl-gadget/bcck8s.go

```diff
@@ -311,6 +315,10 @@ func bccCmd(subCommand, bccScript string) func(*cobra.Command, []string) {
 			}(node.Name, i) // node.Name is invalidated by the above for loop, causes races
 		}
 
+		if params.Node != "" && !nodeFound {
+			contextLogger.Fatalf("Invalid filter: Node %q does not exist", params.Node)
+		}
+
```

Comment:
The same check is done for the non-BCC based gadget in [utils/trace](https://github.com/kinvolk/inspektor-gadget/blob/4a06ad3eb2ec53a321319c24dcc6e5ed7829ec8e/cmd/kubectl-gadget/utils/trace.go#L217-L219), what about moving it to the [PersistentPreRunE function](https://github.com/kinvolk/inspektor-gadget/blob/4a06ad3eb2ec53a321319c24dcc6e5ed7829ec8e/cmd/kubectl-gadget/utils/flags.go#L98-L142) which is already validating other parameters and is added to all the sub-commands that want to support the common flags ([--node](https://github.com/kinvolk/inspektor-gadget/blob/4a06ad3eb2ec53a321319c24dcc6e5ed7829ec8e/cmd/kubectl-gadget/utils/flags.go#L158-L163) is one of them). In this way, we would avoid duplicated code for BCC and non-BCC based gadgets.

Thoughts?

Reply:
Yes, I think it makes sense to have this validation there.